### PR TITLE
Feature/9 start api

### PR DESCRIPTION
**Pull request by MaximilianV**
In this PR, the back-end api is introduced:
- CRUD operations for `ResourceType` and `ResourceInstance`
- The command `npm run swagger` will create the latest API docs in `src/swagger.json`
- Also, the API is documented in the Postman workspace

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,12 +7,15 @@
     "lint": "tslint -p tsconfig.json",
     "development": "npm run lint && node --inspect=9229 -r tsconfig-paths/register -r ts-node/register src/app.ts",
     "build": "tsc && cp src/swagger.json build/swagger.json",
-    "production": "node -r tsconfig-paths/register build/app.js"
+    "production": "node -r tsconfig-paths/register build/app.js",
+    "swagger": "swagger-jsdoc -d swaggerConf.js -o src/swagger.json"
   },
   "dependencies": {
+    "@types/jsonapi-serializer": "^3.6.1",
```
Comment (friedow) [inline package.json]: `@types` are dev dependencies

File: src/app.ts [hunk 2]
```diff
@@ -4,21 +4,40 @@ import bodyParser from 'body-parser';
...
+      '415',
+      'Unsupported Media Type',
+      `Only requests specifying '${contentType}' as content type are allowed.`,
+    ));
+  }
+  next();
+}
 
-async function startApiServer() {
+async function startApiServer(): Promise<void> {
```
Comment (friedow) [inline src/app.ts:28]: 😊👍

File: src/models/ResourceInstance.ts [hunk 3]
```diff
@@ -62,6 +85,15 @@ export class ResourceInstance extends Typegoose {
   // endregion
 
   // region public methods
+  @instanceMethod
+  public updateFromObject(updateObject: any) {
```
Comment (friedow) [inline src/models/ResourceInstance.ts]: Is it possible to enforce ResourceInstance as parameter type here?

File: src/models/ResourceType.ts [hunk 4]
```diff
@@ -15,7 +11,54 @@ export interface Attribute {
...
+@pre<ResourceType>('remove', async function(): Promise<void> {
+  const childTypesCount = await ResourceTypeModel.countDocuments({ parentType: this._id });
+  if (childTypesCount > 0) {
+    return new Promise((resolve, reject) => {
+      reject(new Error(`There are ${childTypesCount} resource types with '${this.name}' as parent. ` +
+        `Type can not be deleted.`));
+    });
+  }
+
+  const instancesCount = await ResourceInstanceModel.countDocuments({ resourceType: this._id });
```
Comment (friedow) [inline src/models/ResourceType.ts:23]: Is this really the intended behavoir? Maybe the user wants to delete a resource type including all of its instances.

File: src/models/ResourceType.ts [hunk 5]
```diff
@@ -43,10 +86,18 @@ export class ResourceType extends Typegoose {
   // endregion
 
   // region public methods
+  @instanceMethod
+  public updateFromObject(updateObject: any) {
+    ResourceTypeModel.schema.eachPath( (path) => {
```
Comment (friedow) [inline src/models/ResourceType.ts]: Arrow function is a duplicate to `ResourceInstance.ts`, extract that code into an own function.

**Review by friedow — CHANGES_REQUESTED** [review]
Well done, clean code 👍

File: src/models/ResourceType.ts (hunk 4 above)
Comment (MaximilianV) [reply src/models/ResourceType.ts:23]: Maybe we can add this functionality later with an additional confirmation, like "This type has 243 instances. Do you really want to delete all instances and the type?". At the moment, I think its easier and safer to allow deletion only if the type is "empty". We of course could think about a route deleting all instances of a type (bulk deleting by property/attribute).

**Review by friedow — APPROVED** [review]

**Pull request merged**